### PR TITLE
Builder

### PR DESCRIPTION
**Pull request by HorjuRares**
Please review my implementation of builder DP, @andreimihalcea97.

### CODE REVIEW

File: Builder/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+add_subdirectory(src)
+
+add_executable(Builder "builder.cpp")
+
+target_link_libraries(Builder PUBLIC CodeBuilder)
```
Comment (andreimihalcea97) [inline Builder/CMakeLists.txt:6]: Why are the includes missing from the CMakelists?

Comment (HorjuRares) [reply Builder/CMakeLists.txt:6]: They're not, they are included in CodeBuilder in src/CMakeLists.txt.

File: Builder/src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+add_library(CodeBuilder STATIC "CodeBuilder.cpp")
```
Comment (andreimihalcea97) [inline Builder/src/CMakeLists.txt]: I would delete this file and move target_include_directories in the CMakeLists from Builder folder.

Comment (HorjuRares) [reply Builder/src/CMakeLists.txt]: Ok.

Comment (HorjuRares) [reply Builder/src/CMakeLists.txt]: Modified according to your indications.

File: Builder/include/CodeBuilder.hpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#include <vector>
+#include <ostream>
+#include <sstream>
+#include <iostream>
+
+#ifndef CODESNIPPET_H
+#define CODESNIPPET_H
+class CodeSnippet
+{
+	std::string class_name;
```
Comment (andreimihalcea97) [inline Builder/include/CodeBuilder.hpp:13]: It's more readable if you place the private members after the public ones and also use the private keyword.

File: Builder/include/CodeBuilder.hpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+	friend class CodeBuilder;
+};
+#endif
+
+#ifndef CODEBUILDER_H
+#define CODEBUILDER_H
+class CodeBuilder
+{
+	CodeSnippet root;
```
Comment (andreimihalcea97) [inline Builder/include/CodeBuilder.hpp:40]: Same here.

**Review by andreimihalcea97 — APPROVED** [review]

**Pull request merged**